cosmos/dbt/graph.py

```diff
@@ -470,9 +477,9 @@ def load_via_dbt_ls_without_cache(self) -> None:
                 self.nodes = nodes
                 self.filtered_nodes = nodes
 
-            if self.project.partial_parse and self.cache_dir:
+            if self.should_use_partial_parse_cache():
                 partial_parse_file = get_partial_parse_path(tmpdir_path)
-                if partial_parse_file.exists():
+                if partial_parse_file.exists() and self.cache_dir:
```

Comment:
```suggestion
                if partial_parse_file.exists():
```
based on Pankaj's comment above that we already check this prior to getting here.

Reply:
Resolving this as https://github.com/astronomer/astronomer-cosmos/pull/1070#discussion_r1657159419